### ISSUE TITLE
Editor unresponsive when removing SoftBody node

### ISSUE THREAD

**Issue by timothyqiu** [godotengine/godot#34650]
**Godot version:** v3.2.beta.custom_build.318c69351

**OS/device including version:** macOS Catalina 10.15.1 / MacBook Pro (Retina, 13-inch, Early 2015)

**Issue description:**

When I delete a `SoftBody` node, the editor becomes unresponsive. It looks like an infinite loop, printing tons of the following error:
```
ERROR: _disconnect: Attempt to disconnect signal 'node_removed' while in emission callback. Use CONNECT_DEFERRED (to be able to safely disconnect) or CONNECT_ONESHOT (for automatic disconnection) as connection flags.
   At: core/object.cpp:1522.
```

**Steps to reproduce:**
1. Create a new 3D scene
2. Add a `SoftBody` node as the child of the default `Spatial` node
3. Assign the `Spatial` node to the `SoftBody`'s `Parent Collision Ignore` property
4. Right click the `SoftBody` and select "Delete Node(s)"
5. Click "OK" when prompted

---

**Update:** I just tried the same thing with `InterpolatedCamera`. Assign the parent Spatial node to its `Target` property and try to delete the node, the editor became unresponsive too. (with the same error output.)

**Comment by akien-mga** [comment]
This should be slightly easier to debug after #34753 which improves the error message.

**Comment by timothyqiu** [comment]
I think I located the cause.

The `SceneTreeEditor` connects to "node_removed" signal of the tree on enter tree, and disconnects on exit.
https://github.com/godotengine/godot/blob/9d3424f61ddd1e6bec78381aa4a0c6acec53d290/editor/scene_tree_editor.cpp#L665-L683

Once a `SceneTreeEditor` is created & opened by clicking the "Assign..." button, it's connected to the `node_removed` signal of the edited scene tree. This `SceneTreeEditor` instance is alive as long as the inspected node is still selected.

Once the inspected node is removed  (deleting the root node in my case, and switching the scene as in #34769), the scene tree emits the "node_removed" signal, the inspector will then be cleared, and propagates to the exit tree notification code above.

Thus, it is trying to disconnect the "node_removed" signal of the scene tree inside the scene tree's signal emission process. 

I tried to change the connect call to `CONNECT_DEFERRED`,  but it crashes the editor. Likely because when the deferred `node_removed` call is finally executed, the node object in the argument has already been destroyed.
